src/bumper.cpp

```diff
@@ -0,0 +1,103 @@
+#include "bumper.h"
+
+Bumper::Bumper()
+{
+}
+
+Bumper::~Bumper()
+{
+
+}
+
+void Bumper::setRobotRadius(double width, double length)
+{
+    _robotRadiusWidth = width;
+    _robotRadiusLength = length;
+}
+
+void Bumper::setRobotRadius(double radius)
+{
+    _robotRadiusLength = radius;
+    _robotRadiusWidth = radius;
+}
+
+void Bumper::setBumperRadius(double size)
+{
+    _bumperRadius = size;
+}
+
+void Bumper::generateBumperData(const World& world, const Robot& robot, std_msgs::Bool& scan_msg_f, std_msgs::Bool& scan_msg_r) const
+{
+    // Generate laser data to perform bumper check
+    sensor_msgs::LaserScan lrf_msg;
+    LRF::generateLaserData(world, robot, lrf_msg);
+
+    bool hitF = false;
+    bool hitR = false;
+
+    int N = lrf_msg.ranges.size();
+    for(int i = 0; i<lrf_msg.ranges.size(); i++)
+    {
+        double theta_i = lrf_msg.angle_min + i*lrf_msg.angle_increment;
+
+        // If range larger than bumper radius do nothing
+        if(lrf_msg.ranges[i] > _bumperRadiusTheta(theta_i))
+        {
+            continue;
+        }
+        if(lrf_msg.ranges[i]< lrf_msg.range_min)
+        {
+            continue;
+        }
+        // If range smaller than bumper radius, and is behind the robot
+        if (_isRear(theta_i))
+        {
+            hitR = true;
+            continue;
+        }
+
+        // If range smaller than bumper radius, and is in front of the robot
+        if (_isFront(theta_i))
+        {
+            hitF = true;
+            continue;
+        }
+    }
+    // Write variables to message 
+    scan_msg_f.data = hitF;
+    scan_msg_r.data = hitR;
+}
+
+double Bumper::_radiusTheta(const double theta) const
+{
+    double ca = cos(theta)/_robotRadiusWidth;
+    ca = ca*ca;
+
+    double sb = sin(theta)/_robotRadiusLength;
+    sb = sb*sb;
+
+    double radius = 1/sqrt(ca + sb);
+    return radius; 
+}
+
+double Bumper::_bumperRadiusTheta(const double theta) const
+{
+    return _radiusTheta(theta) + _bumperRadius;
+}
```

Comment:
Waarom heb je deze twee functies nodig? Kunnen we niet gewoon zeggen dat de robot rond is?